Improve `__repr__` methods of some classes beeware#432 (#436)
beeware/rubicon-objc: changes/436.misc.rst

Reviewer: This is a significant feature change - so it should be `432.feature.rst`, referencing the issue number that describes the new feature.

> Improved `__repr__` methods of `ObjCBoundMethod`, `ObjCClass`, `ObjCInstance`, `ObjCMethod`, `ObjCPartialMethod`, and `ObjCProtocol`.

Reviewer: ```suggestion
The `__repr__` output for  `ObjCBoundMethod`, `ObjCClass`, `ObjCInstance`, `ObjCMethod`, `ObjCPartialMethod`, and `ObjCProtocol` was simplified.
```
